### PR TITLE
edit: Windows fixes

### PR DESCRIPTION
**Pull request by 0xmichalis**
Contains the following fixes for Windows users of kubectl edit:
- Defaults to notepad as the default Windows editor (https://github.com/openshift/origin/issues/3062 -> https://github.com/openshift/origin/pull/3086)
- Uses CRLF line endings (https://github.com/openshift/origin/issues/4904 -> https://github.com/openshift/origin/pull/5300)
- Ensures a file lock is freed (https://github.com/openshift/origin/issues/4904 -> https://github.com/openshift/origin/pull/5306)

@kubernetes/kubectl @JanetKuo

@smarterclayton @fabianofranz fyi

### CODE REVIEW

File: pkg/util/crlf.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import (
+	"bytes"
+	"io"
+)
+
+type crlfWriter struct {
+	io.Writer
+}
+
+func NewCRLFWriter(w io.Writer) io.Writer {
```
Comment (smarterclayton) [inline pkg/util/crlf.go]: Needs Godoc

File: pkg/kubectl/cmd/util/editor/editor.go [hunk 2]
```diff
@@ -133,6 +143,7 @@ func (e Editor) LaunchTempFile(prefix, suffix string, r io.Reader) ([]byte, stri
 		os.Remove(path)
 		return nil, path, err
 	}
+	f.Close()
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/editor/editor.go]: Add a comment here indicating why we close

**Comment by smarterclayton** [comment]
This is a faithful conversion of the downstream changes.

**Comment by jlowdermilk** [comment]
Other than missing comments, lgtm

**Comment by kargakis** [comment]
comments addressed

**Comment by kargakis** [comment]
had to rebase

**Pull request merged**